### PR TITLE
[Shapes] Bug fix for getting the wrong backgroundColor when shapeGenerator is set.

### PR DESCRIPTION
**Pull request by yarneo**
This bug fix resolves issue #5126 

Quoting issue:
"Shape-able components such as Buttons and Chips rely on an `MDCShapedShadowLayer` as their main layer class underlying the view. Due to that, there is new logic for setting and getting the `backgroundColor` and is done through `self.layer.shapedBackgroundColor`. There exists logic in these components for setting the `shapedBackgroundColor` when calling the `backgroundColor` setter. However, when fetching `backgroundColor`, then Button and Chips don't have a getter to fetch `shapedBackgroundColor`. This means that when those components do have a custom shape active, then when fetching the background color, it will not return the correct one."

### CODE REVIEW

File: components/Buttons/src/MDCButton.m [hunk 1]
```diff
@@ -536,6 +536,10 @@ - (void)setBackgroundColor:(nullable UIColor *)backgroundColor {
   [self updateBackgroundColor];
 }
 
+- (UIColor *)backgroundColor {
```
Comment (galiak11) [inline components/Buttons/src/MDCButton.m:539]: What's the difference between backgroundColorForState and backgroundColor in MDCButton?  In what cases is it preferred to call one over the other?

File: components/Buttons/tests/unit/ButtonsShapeThemerTests.m [hunk 2]
```diff
@@ -56,4 +57,10 @@ - (void)testMDCButtonShapeThemer {
                         self.shapeScheme.smallSurfaceShape.bottomRightCorner);
 }
 
+- (void)testBackgroundColorAfterButtonTheming {
+  MDCButtonScheme *scheme = [[MDCButtonScheme alloc] init];
+  [MDCTextButtonThemer applyScheme:scheme toButton:self.button];
+  XCTAssertEqualObjects([UIColor clearColor], self.button.backgroundColor);
+}
+
```
Comment (galiak11) [inline components/Buttons/tests/unit/ButtonsShapeThemerTests.m]: If I understand correctly, the original issue was getting the background color after having a custom shape active. Can we add a test to cover that scenario (if it's not already covered by existing test(s)).

File: components/Buttons/src/MDCButton.m (hunk 1 above)
Comment (yarneo) [reply components/Buttons/src/MDCButton.m:539]: backgroundColorForState provides the backgroundColor for a specific state, if it be Selected, Normal, Highlighted. backgroundColor is the default getter for UIViews providing the current backgroundColor.

File: components/Buttons/tests/unit/ButtonsShapeThemerTests.m (hunk 2 above)
Comment (yarneo) [reply components/Buttons/tests/unit/ButtonsShapeThemerTests.m]: done.

File: components/Buttons/src/MDCButton.m (hunk 1 above)
Comment (andrewoverton) [reply components/Buttons/src/MDCButton.m:539]: In UIButton there are a whole bunch of readonly properties that do this, but they have a `current` prefix. I guess it's probably not worth a breaking change to keep that convention here, huh.

Comment (andrewoverton) [reply components/Buttons/src/MDCButton.m:539]: Oh you're overriding UIView's backgroundColor huh

**Review by andrewoverton — APPROVED** [review]

**Pull request merged**